sys/net/gnrc/network_layer/sixlowpan/gnrc_sixlowpan.c

```diff
@@ -327,6 +335,7 @@ static void _send(gnrc_pktsnip_t *pkt)
                 fbuf->datagram_size = datagram_size;
                 fbuf->tag = gnrc_sixlowpan_frag_fb_next_tag();
                 fbuf->offset = 0;
+                fbuf->best_frag_size = 0;
```

Comment:
not sure why that's needed

Reply:
When I tested, I did not get here so I just initielized the field to zero.
But it could be that we need to query `NETOPT_MAX_SDU_SIZE` here as well.
However let´s say module `gnrc_sixlowpan_frag_hint` and `gnrc_sixlowpan_frag_minfwd` are used, then we may take the path `gnrc_sixlowpan_iphc_send()` --> `gnrc_sixlowpan_frag_minfwd_frag_iphc()`, where we check
```C
    if (!ipv6_addr_is_link_local(ipv6_dst) &&
        (fbuf->datagram_size > netif->sixlo.max_frag_size)) {
        fbuf->pkt = pkt;    /* packet might have been rewritten */
        /* put slack of IPHC in first fragment */
        fbuf->hint.fragsz = pkt->next->size;
        fbuf->hint.fragsz_uncomp = orig_datagram_size -
                                       gnrc_pkt_len(pkt->next->next);
        gnrc_sixlowpan_frag_send(NULL, fbuf, 0);
        res = 0;
    }
```
We should probably not compare `datagram_size` with `max_frag_size` because it could result in false fragmentation. But I am not sure if it is supposed to be like that, or what `fragsz` is used for, nor why there are two points where a fragemnt buffer is allocated. If I knew what was used for I could test if it still works.
But so far I have not read RFC 8930, nor have I tested yet what exactly happens when we land here.
I will do that in a moment.

Reply:
Ok `frag_minfwd` and `frag_sfr` must be adapted too.

Reply:
Yea that's why I found it easier to just re-use `max_frag_size` but I didn't check if it's really not used for anything else.